content/general/concepts/algorithm/terms/euclidean-algorithm/kadane-algorithm.md

```diff
@@ -0,0 +1,107 @@
+---
+Title: 'Kadane's Algorithm'
+Description: 'Returns the maximum sum of a subarray within an array of numbers'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Algorithms'
+  - 'Arithmetic'
+  - 'Arrays'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**`Kadane's algorithm`** is a dynamic programming approach utilized to identify the maximum sum of a subarray within a given array of numbers. The algorithm solves this problem efficiently in a single pass through the array by maintaining two variables: `maxEndingHere` and `maxSoFar`. The algorithm iterates through the array, updating these variables at each step to track the maximum subarray sum ending at the current element and the overall maximum subarray sum found so far. The final value of maxSoFar represents the maximum subarray sum in the given `array`. This algorithm guarantees a time complexity of `O(n)`, making it an optimal solution for this problem.
+
+## Explanation
+
+**Here's how Kadane's algorithm can be applied to a problem:**
+
+- Start with the first number and add it to a running total. Then, move to the next number and determine whether to continue with the current sum or start a new sum from this number. Always aim to maintain the largest sum seen so far.
+- For example, consider the array {-2, -1, -3, 4, -1, 2, 1, -5, 4}. Initialize two integer variables: one named `maxEndingHere` and the other called `maxSoFar`. These variables serve to track the maximum subarray sum ending at the current position and the overall maximum subarray sum.
+- The loop commence with the first element, which is -2. So, both `maxEndingHere` and `maxSoFar` start at -2.
+- As the loop proceed to the next element, which is -1, compare it to the sum of the current element and the previous value of `maxEndingHere`. Update `maxEndingHere` to **-1** because it's greater than the sum of **-2** and **-1**. Also, compare `maxSoFar` with the updated value of `maxEndingHere`. Since **-2** is less than **-1**, update `maxSoFar` to **-1**.
```

Comment:
```suggestion
Proceed to the next element, -1, and compare it to the sum of itself and the previous value of `maxEndingHere`.  Since -1 is greater than the sum of -2 and -1, set`maxEndingHere` to  -1. Also, compare `maxSoFar` with the updated value of `maxEndingHere`. Since -2 is less than -1 set `maxSoFar` to  -1.
```